x/tx/signing/get_signers.go

```diff
@@ -175,3 +187,37 @@ func (c *GetSignersContext) GetSigners(msg proto.Message) ([]string, error) {
 
 	return f(msg), nil
 }
+
+// GetSignersForAny returns the signers for a given proto message wrapped inside
+// an Any. It can be used to retrieve the signers for a message which does not
+// have a concrete protoreflect type in the current binary because it uses dynamicpb
+// under the hood. This is useful for usage in dynamic clients (like autocli) or
+// for working with legacy gogo protobuf messages.
+func (c *GetSignersContext) GetSignersForAny(any *anypb.Any) ([]string, error) {
+	url := any.TypeUrl
+	typ, err := c.protoTypes.FindMessageByURL(url)
+	if err != nil {
+		if err == protoregistry.NotFound {
+			msgName := protoreflect.FullName(url)
+			if i := strings.LastIndexByte(url, '/'); i >= 0 {
+				msgName = msgName[i+len("/"):]
+			}
```

Comment:
I was working on similar code yesterday, in https://github.com/cosmos/cosmos-sdk/pull/15302.

For this specific lines, I simply used `strings.TrimPrefix("/")`, assuming that the whole cosmos ecosystem uses `/`-prefix any type urls

Reply:
Should we make a util function for this?